warmrestart/warm_restart.cpp

```diff
@@ -1,108 +1,139 @@
 #include <string>
+#include <climits>
 #include "logger.h"
 #include "schema.h"
-#include <climits>
 #include "warm_restart.h"
 
 namespace swss {
 
 const WarmStart::WarmStartStateNameMap WarmStart::warmStartStateNameMap =
 {
-    {INIT,          "init"},
+    {INITIALIZED,   "initialized"},
     {RESTORED,      "restored"},
     {RECONCILED,    "reconciled"}
 };
 
-WarmStart &WarmStart::getInstance()
+WarmStart &WarmStart::getInstance(void)
 {
     static WarmStart m_warmStart;
     return m_warmStart;
 }
 
+/*
+ * WarmStart's initialization method -- to be invoked once per application.
+ */
+void WarmStart::initialize(const std::string &app_name,
+                           const std::string &docker_name)
+{
+    auto& warmStart = getInstance();
+
+    if (warmStart.m_initialized)
+    {
+        return;
+    }
+
+    warmStart.m_stateDb =
+        std::make_shared<swss::DBConnector>(STATE_DB, swss::DBConnector::DEFAULT_UNIXSOCKET, 0);
+
+    warmStart.m_stateWarmRestartTable =
+        std::unique_ptr<Table>(new Table(warmStart.m_stateDb.get(), STATE_WARM_RESTART_TABLE_NAME));
+
+    warmStart.m_cfgDb =
+        std::make_shared<swss::DBConnector>(CONFIG_DB, swss::DBConnector::DEFAULT_UNIXSOCKET, 0);
+
+    warmStart.m_cfgWarmRestartTable =
+        std::unique_ptr<Table>(new Table(warmStart.m_cfgDb.get(), CFG_WARM_RESTART_TABLE_NAME));
+
+    setWarmStartState(app_name, WarmStart::INITIALIZED);
```

Comment:
For portsyncd and vlanmgrd,  since no reconciliation is needed,  "setWarmStartState("orchagent", WarmStart::INITIALIZED);" was not called explicitly like that for orchagent.
Here setWarmStartState() is called implicitly, VS test check for portsyncd and vlanmgrd will fail.  

I think you may either call setWarmStartState() for portsyncd and vlanmgrd at appropriate places
```
    if (WarmStart::isWarmStart())
    {
           WarmStart::setWarmStartState("portsyncd", WarmStart::RECONCILED);
    }
```
or take other fixes.

Reply:
Are you saying that you want to transition these two modules to RECONCILED state even though there's no reconciliation phase for them? That sounds a bit strange, isn't it?

Reply:
Current vs test for portsyncd and vlanmgrd warm restart is failing due to changes here,  it should be fixed.  There are multiple alternatives to the problem, please take the one you think the most appropriate.